cmd/setup.go

```diff
@@ -971,26 +978,63 @@ func configureSite(conf map[string]interface{}, loadpoints []*core.Loadpoint, ta
 	return site, nil
 }
 
-func configureLoadpoints(conf globalconfig.All) ([]*core.Loadpoint, error) {
-	if len(conf.Loadpoints) == 0 {
-		return nil, errors.New("missing loadpoints")
+func configureLoadpoints(conf globalconfig.All) error {
+	for id, cc := range conf.Loadpoints {
+		cc.Name = "lp-" + strconv.Itoa(id+1)
+
+		log := util.NewLoggerWithLoadpoint(cc.Name, id+1)
+		settings := coresettings.NewDatabaseSettingsAdapter(fmt.Sprintf("lp%d.", id+1))
+
+		instance, err := core.NewLoadpointFromConfig(log, settings, cc.Other)
+		if err != nil {
+			return fmt.Errorf("failed configuring loadpoint: %w", err)
+		}
+
+		if err := config.Loadpoints().Add(config.NewStaticDevice(cc, loadpoint.API(instance))); err != nil {
+			return &DeviceError{cc.Name, err}
+		}
 	}
 
-	var loadpoints []*core.Loadpoint
+	// append devices from database
+	configurable, err := config.ConfigurationsByClass(templates.Loadpoint)
+	if err != nil {
+		return err
+	}
 
-	for id, cfg := range conf.Loadpoints {
-		log := util.NewLoggerWithLoadpoint("lp-"+strconv.Itoa(id+1), id+1)
-		settings := &core.Settings{Key: "lp" + strconv.Itoa(id+1) + "."}
+	for _, conf := range configurable {
+		cc := conf.Named()
 
-		lp, err := core.NewLoadpointFromConfig(log, settings, cfg)
+		id := len(config.Loadpoints().Devices())
+		name := "lp-" + strconv.Itoa(id+1)
+		log := util.NewLoggerWithLoadpoint(name, id+1)
+
+		settings := coresettings.NewConfigSettingsAdapter(log, &conf)
+
+		dynamic, static, err := loadpoint.SplitConfig(cc.Other)
 		if err != nil {
-			return nil, fmt.Errorf("failed configuring loadpoint: %w", err)
+			return fmt.Errorf("failed configuring loadpoint: %w", err)
 		}
 
-		loadpoints = append(loadpoints, lp)
+		instance, err := core.NewLoadpointFromConfig(log, settings, static)
+		if err != nil {
+			return fmt.Errorf("failed configuring loadpoint: %w", err)
+		}
+
+		if err := dynamic.Apply(instance); err != nil {
```

Comment:
@andig hier (und in der Methode drüber) wird für jede Eigenschaft, die am LP beim Initialisieren gesetzt wird in die DB geschrieben (SELECT+UPDATE). Für die Funktionsweise ist das mEn ein Problem, aber unnötig und unerwartet. Beim Initialisieren würde ich keine Schreiboperationen erwarten. Liegt halt daran, wie wir mit "Settings" und den Settern umgehen.